mender-auth/ipc/server_test.cpp

```diff
@@ -138,40 +167,27 @@ TEST_F(ListenClientTests, TestListenFetchJWTToken) {
 	config.server_url = "http://127.0.0.1:" TEST_PORT_2;
 	config.tenant_token = "dummytenanttoken";
 
-	ipc::type::webserver::Caching server {loop, config};
-	server.Cache("NotYourJWTTokenBitch", "http://127.0.0.1:" TEST_PORT_2);
-
-	const string server_url {"http://127.0.0.1:" TEST_PORT};
-
-	err = server.Listen(server_url, "./private-key.rsa.pem", test_device_identity_script);
+	ipc::Server server {loop, config};
+	server.Cache("NotYourJWTTokenBitch", "http://127.1.1.1:" TEST_PORT_2);
+	err = server.Listen("./private-key.rsa.pem", test_device_identity_script);
 	ASSERT_EQ(err, error::NoError);
 
 	// Set up the test client (Emulating mender-update)
-	http::ClientConfig test_client_config {};
-	http::Client update_client(test_client_config, loop);
-	auto req = make_shared<http::OutgoingRequest>();
-	req->SetMethod(http::Method::GET);
-	const string url {"http://127.0.0.1:" TEST_PORT "/fetchjwttoken"};
-	req->SetAddress(url);
-	update_client.AsyncCall(
-		req,
-		[](http::ExpectedIncomingResponsePtr exp_resp) {
-			ASSERT_TRUE(exp_resp) << exp_resp.error().message;
-		},
-		[](http::ExpectedIncomingResponsePtr exp_resp) {
-			ASSERT_TRUE(exp_resp) << exp_resp.error().message;
-			ASSERT_EQ(exp_resp.value()->GetStatusCode(), 200);
+	events::Timer timer {loop};
+	dbus::DBusClient client {loop};
+	err = client.CallMethod<expected::ExpectedBool>(
+		"io.mender.AuthenticationManager",
+		"/io/mender/AuthenticationManager",
+		"io.mender.Authentication1",
+		"FetchJwtToken",
+		[&loop, &timer](expected::ExpectedBool ex_value) {
+			ASSERT_TRUE(ex_value) << ex_value.error().message;
+			EXPECT_TRUE(ex_value.value());
+			timer.AsyncWait(chrono::seconds {1}, [&loop](error::Error err) { loop.Stop(); });
```

Comment:
Why do we use a timer here? Can't we wait for the signal instead?

Reply:
We can. I just replaced things in the original test that used a separate thread for stopping the loop after a second. I guess I was happy enough with the improvement I made! :joy: